File: C-lang/check-password.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

int main() {
    bool ispresent = false; 
    char string[30];
    int i;

    printf("Enter a string: ");
    scanf("%s", string);

    for (i = 0; string[i] != '\0'; i++) {
        if (string[i] >= '0' && string[i] <= '9') {
            ispresent = true;
            break; // no need to continue
        }
    }

    if (ispresent == true) {
        printf("Number is present in password");
    } else {
        printf("No number in password");
    }

    return 0;
}

File: Python/python-practice/check-password.py

string = str(input("Enter your password : "))

isNumberPresent = False
isPresentSpecialChar = False
isCharPresent = False

for i in range(len(string)):
    if(string[i] >= '0' and string[i] <= '9'):
        isNumberPresent = True

    if(string[i] == '#' or string[i] == '@' or string[i] == '$' or string[i] == '%'):
        isPresentSpecialChar = True

    if(string[i] >= 'a' and string[i] <= 'z' or string[i] >= 'A' and string[i] <= 'Z'):
        isCharPresent = True


if(isPresentSpecialChar == True and isNumberPresent == True and isCharPresent == True):
    print("Password is strong, number and special character is present")

elif(isPresentSpecialChar == True):
    print("Password is weak, special character is present")

elif(isNumberPresent == True):
    print("Password is weak, number is present")

elif(isCharPresent == True):
    print("Password is strong, character is present")

else:
    print("Password is weak")